include/multipass/cli/command.h

```diff
@@ -148,13 +149,37 @@ class Command
     {
         using SuccessCallableTraits = multipass::callable_traits<SuccessCallable>;
         using FailureCallableTraits = multipass::callable_traits<FailureCallable>;
-        using FailureCallableArgType = typename FailureCallableTraits::template arg<0>::type;
+        using SuccessCallableArg0Type = std::remove_reference_t<typename SuccessCallableTraits::template arg<0>::type>;
+        using FailureCallableArg0Type = std::remove_reference_t<typename FailureCallableTraits::template arg<0>::type>;
 
-        static_assert(SuccessCallableTraits::num_args == 1, "");
-        static_assert(FailureCallableTraits::num_args == 1, "");
-        static_assert(std::is_same<typename SuccessCallableTraits::return_type, ReturnCode>::value, "");
-        static_assert(std::is_same<typename FailureCallableTraits::return_type, ReturnCode>::value, "");
-        static_assert(std::is_same<FailureCallableArgType, grpc::Status&>::value, "");
+        static_assert(std::is_same<typename SuccessCallableTraits::return_type, ReturnCode>::value);
+        static_assert(std::is_same<typename FailureCallableTraits::return_type, ReturnCode>::value);
+
+        static_assert(SuccessCallableTraits::num_args == 1);
+        static_assert(std::is_base_of_v<google::protobuf::Message, SuccessCallableArg0Type>,
+                      "`on_success` doesn't receive a Message");
```

Comment:
Would this be clearer?
```suggestion
                      "`on_success` should receive a Message");
```

Reply:
Sure, I'll change that.